ui/packages/catalog-ui-search/src/main/webapp/component/sharing/sharing.view.js

```diff
@@ -0,0 +1,328 @@
+            width: '100%',
+            bottom: '130px',
+          }}
+        >
+          {this.state.items.map((item, i) => {
+            return (
+              <div key={item.id} style={{ margin: '0 50px' }}>
+                <div style={{ display: 'inline-block', width: '50%' }}>
+                  <span
+                    className={
+                      item.type === 'user' ? 'fa fa-user' : 'fa fa-users'
+                    }
+                  />
+                  {item.type === 'user' ? (
+                    <Text
+                      style={{
+                        display: 'inline-block',
+                        padding: '5px',
+                        width: '80%',
+                      }}
+                      value={item.value}
+                      placeholder="Enter a user's email"
+                      showLabel={false}
+                      onChange={value => this.handleChangeInput(i, value)}
+                    />
+                  ) : (
+                    <span style={{ marginLeft: '12px' }}> {item.value} </span>
+                  )}
+                </div>
+                <Enum
+                  style={{ display: 'inline-block', width: 'calc(50% - 70px)' }}
+                  options={
+                    item.type === 'user'
+                      ? [
+                          { label: 'No Access', value: 'none' },
+                          { label: 'Read Only', value: 'read' },
+                          { label: 'Read and Write', value: 'write' },
+                          { label: 'Read, Write, and Share', value: 'share' },
+                        ]
+                      : [
+                          { label: 'No Access', value: 'none' },
+                          { label: 'Read Only', value: 'read' },
+                          { label: 'Read and Write', value: 'write' },
+                        ]
+                  }
+                  value={item.access}
+                  showLabel={false}
+                  onChange={value => this.handleChangeSelect(i, value)}
+                />
+                {item.type === 'user' && (
+                  <button
+                    style={{
+                      display: 'inline-block',
+                      width: '50px',
+                      verticalAlign: 'middle',
+                    }}
+                    onClick={() => {
+                      this.remove(i)
+                    }}
+                    className="is-negative"
+                  >
+                    <span className="fa fa-minus" />
+                  </button>
+                )}
+              </div>
+            )
+          })}
+        </div>
+        <div
+          style={{
+            position: 'absolute',
+            bottom: '20px',
+            left: '20px',
+            width: 'calc(100% - 40px)',
+          }}
+        >
+          <button
+            style={{ width: '100%', marginBottom: '10px' }}
+            onClick={() => {
```

Comment:
✏️ `onClick = { this.add }`

Reply:
Good catch, updated my comment to reflect that :)

Reply:
`this.add.bind(this)` and `() => { this.add() }` are equivalent I believe, unless you prefer one for style reasons

Reply:
The main reason is less lines